tests/nosetests/pyanaconda_tests/modules/storage/partitioning/module_part_blivet_test.py

```diff
@@ -25,15 +26,18 @@
 from pyanaconda.modules.storage.devicetree import create_storage
 from tests.nosetests.pyanaconda_tests import patch_dbus_publish_object, check_task_creation
 
-from blivetgui.communication.proxy_utils import ProxyID
-
 from pyanaconda.modules.storage.partitioning.blivet.blivet_module import BlivetPartitioningModule
 from pyanaconda.modules.storage.partitioning.blivet.blivet_interface import \
     BlivetPartitioningInterface
 from pyanaconda.modules.storage.partitioning.interactive.interactive_partitioning import \
     InteractivePartitioningTask
 
 
+# blivet-gui is supported on Fedora, but not ELN/CentOS/RHEL
+# define this env to skip the corresponding tests
+NO_BLIVET_GUI = os.getenv('NO_BLIVET_GUI')
```

Comment:
I could argue that this solution has the same issues as the `ImportError` solution. If we mess up the environment (and we will definitely do that one day), we will start to skip these tests without noticing. Therefore, the tests should run based on the importability of `blivetgui`, but we can have a better control over the skipped tests in general.

This kind of error will affect the coverage of the code. I am planning to enable `codecov.io` anyway, so any change in skipped tests will be visible from every pull request.

More strict solution would be to work with a number of skipped tests and verify that we have skipped exactly that number of tests. I don't think it is necessary, but at least it is more universal and not specific for blivetgui.

Reply:
I'd argue that it's much harder to "accidentally" set `$NO_BLIVET_GUI` in our container than blivet-gui suddenly breaking in Rawhide. The former is "explicitly opt in" (and if you do nothing, the test fails), the latter (or any approach with catching `ImportError`) is "if you do nothing, the test wrongly passes".

> verify that we have skipped exactly that number of tests

That has the same principal problem, though -- you need to control *that* condition somehow, based on whether you are in a Rawhide or ELN container.

So, I certainly appreciate better ideas how to tighten the control over the skips. Gating on reduced code coverage certainly sounds like a great idea!

But I strongly veto an `except ImportError:` approach, as that's just laying a trap for ourselves.

Reply:
> I'd argue that it's much harder to "accidentally" set $NO_BLIVET_GUI in our container than blivet-gui suddenly breaking in Rawhide. 

I like your optimism. Anyway, if blivet-gui is broken, is it really our problem? Do we really want to block anaconda because of that? It should be possible to finish the installation with broken blivet-gui. Therefore, it will not block anyone, but our CI will fail until it is fixed.

Also, blivet-gui is gated in Rawhide ;-)

> That has the same principal problem, though -- you need to control that condition somehow, based on whether you are in a Rawhide or ELN container.

Sure, but it doesn't create additional noise in the unit tests. The number of skipped tests will be kept outside the container and checked outside the container. It is not necessary to propagate this information into the container, where we can accidentally change it.

@jkonecny12, what do you think?

Reply:
> Anyway, if blivet-gui is broken, is it really our problem? Do we really want to block anaconda because of that?

I suppose that's eventually your team's decision. In cockpit's team I'd say "absolutely, yes" - if my project's dependencies don't work, then I'm screwed as well, and want to know about it, report it, and see that it gets fixed. Just covering your eyes and let users figure it out isn't really better IMHO. :grin: 

> Therefore, it will not block anyone, but our CI will fail until it is fixed.

That is a  good point. Note that this PR is mostly just the initial workflow. Once it lands, we want to build the container in a nightly workflow and gate it on tests, so that (1) we don't block our PRs on Rawhide regressions, and (2) make the tests faster. I.e. exactly the same that we do for the normal rawhide anaconda-ci right now.

But aside from that, it's obvious that you don't like this current approach. What would you rather have instead?

Reply:
I thought about this again, and I'm actually not sure what your main objection is: The way we get that information from the outside into the unit test? (I do agree that the current approach is rather inelegant)? Or the fact that we "insist" on testing blivet-gui on Rawide? In our call I had the impression that it was the latter, but we are doing that right now, and always have for rawhide. So that wouldn't be anything new.

But anyway, I'll work on a patch tomorrow where the unit test itself will check if blivet-gui is installed, and drop the whole env var business. The .spec already enforces the installation of  blivet-gui on Rawhide, that's good enough. And it will make the structure much simpler. Thanks for the review!